data/simpy.gram

```diff
@@ -324,10 +324,28 @@ power[expr_ty]:
     | primary 
 primary[expr_ty]:
     | !'await' a=atom !'.' !'(' !'[' { a }
-    | [AWAIT] atom ('.' NAME | '[' slices ']' | '(' [arguments] ')')*
-
-slices: ','.slice+ [',']
-slice: [expression] ':' [expression] [':' [expression]] | expression
+    | AWAIT a=primary_raw { _Py_Await(a, EXTRA) }
+    | primary_raw
+primary_raw[expr_ty]:
+    | a=primary_raw '.' b=NAME { _Py_Attribute(a, ((expr_ty) b)->v.Name.id, Load, EXTRA) }
+    | a=primary_raw '[' b=expression ']' { _Py_Subscript(a, _Py_Index(b, p->arena), Load, EXTRA) }
+    | a=primary_raw '[' b=slice_expressions ']' { _Py_Subscript(a, b, Load, EXTRA) }
+    | a=primary_raw '[' b=slices ']' { _Py_Subscript(a, b, Load, EXTRA) }
```

Comment:
An alternative way to factor this (up to you) is to introduce a sub-rule ("slicing"?) that represents the `[` ... `]` part.

Reply:
Wouldn't this need a helper type, since `expression` returns an `expr_ty`, but `slice_expressions` and `slices` return `slice_ty`? If that's so, I'd leave it as is.

Reply:
I was thinking of this:
```
primary[expr_ty]:
    ...
    | a=primary b=slicing { _Py_Subscript(a, b, Load, EXTRA) }
    ...
slicing[slice_ty]:
    | '[' b=expression ']' { _Py_Index(b, p->arena) }
    | '[' b=slice_expressions ']' { b }
    | '[' b=slices ']' { b }
```